src/Transport/AMQPSender.php

```diff
@@ -81,4 +81,25 @@ public function send(Envelope $envelope): Envelope
 
         return $envelope;
     }
+
+    /**
+     * @throws TransportException
+     * @throws Throwable
+     */
+    #[Override]
+    public function flush(): void
+    {
+        try {
+            $this->retryFactory->retry()
+                ->run(function (Retry $_retry, bool $isRetry): void {
+                    if ($isRetry) {
+                        $this->connection->reconnect();
```

Comment:
If we get an exception on flush, I don't think we can reconnect? The batch data will be lost?